### PR TITLE
Implement vehicle pass signal game command

### PR DESCRIPTION
**Pull request by AaronVanGeffen**

### CODE REVIEW

File: src/OpenLoco/src/GameCommands/VehiclePassSignal.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+using namespace OpenLoco::Interop;
+
+namespace OpenLoco::GameCommands
+{
+    // 0x004B0B50
+    static uint32_t vehiclePassSignal(EntityId headId, const uint8_t flags)
+    {
+        setExpenditureType(ExpenditureType::TrainRunningCosts);
+
+        try
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/GameCommands/VehiclePassSignal.cpp:17]: a try. Interesting whats the thinking behind it?

File: src/OpenLoco/src/GameCommands/VehiclePassSignal.cpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        {
+            Vehicles::Vehicle train(headId);
+            setPosition(train.veh2->position);
+
+            auto& head = train.head;
+            if (!head->canBeModified())
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/GameCommands/VehiclePassSignal.cpp:23]: there is a check missing for the owner to make sure the company is allowed to modify.

File: src/OpenLoco/src/GameCommands/VehiclePassSignal.cpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                return 0;
+            }
+
+            if (head->tileX == -1)
+            {
+                return 0;
+            }
+
+            head->var_5C = 2;
+            train.veh1->var_48 |= 1;
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/GameCommands/VehiclePassSignal.cpp:39]: Its hightime we convert this into an enum flags. Looks like (1<<0) is pass signal judging by all the other users.

File: src/OpenLoco/src/GameCommands/VehiclePassSignal.cpp (hunk 1 above)
Comment (AaronVanGeffen) [reply src/OpenLoco/src/GameCommands/VehiclePassSignal.cpp:17]: Constructing `Train` can throw a std::exception

File: src/OpenLoco/src/GameCommands/VehiclePassSignal.cpp (hunk 2 above)
Comment (AaronVanGeffen) [reply src/OpenLoco/src/GameCommands/VehiclePassSignal.cpp:23]: Oops, I don't know where that got lost. Will re-instate.

File: src/OpenLoco/src/GameCommands/VehiclePassSignal.cpp (hunk 3 above)
Comment (AaronVanGeffen) [reply src/OpenLoco/src/GameCommands/VehiclePassSignal.cpp:44]: Any ideas what (1 << 1) and (1 << 2) mean? Tbh while I agree I'd prefer to handle this in another PR.

Comment (duncanspumpkin) [reply src/OpenLoco/src/GameCommands/VehiclePassSignal.cpp:44]: okay make a follow on.

Comment (AaronVanGeffen) [reply src/OpenLoco/src/GameCommands/VehiclePassSignal.cpp]: Eh, it's less of a diff than I thought. Amended this PR instead.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**